### PR TITLE
NEW add functionnality for more customization of membership

### PR DESCRIPTION
**Pull request by ptibogxiv**
fundations needs more personnalization with subscription creation and more automatism for big fundation
import from my alternative module AdherentPlus in github
- more customization of membership current date to date or fix date ton a fix date (with or without prorata), price, welcome price... (need for integration of  stripe subscription)
-add groupmembership for member of a type (member with submember ex father and their children, with one thirdparty link to father)

### CODE REVIEW

File: htdocs/adherents/type.php [hunk 1]
```diff
@@ -214,7 +228,7 @@
 {
 	//dol_fiche_head('');
 
-	$sql = "SELECT d.rowid, d.libelle as label, d.subscription, d.vote";
+	$sql = "SELECT d.rowid, d.libelle as label, d.subscription, d.vote, d.welcome, d.price, d.vote, d.automatic, d.family";
```
Comment (hregis) [inline htdocs/adherents/type.php]: @ptibogxiv 
"d.vote" is defined twice

Comment (ptibogxiv) [reply htdocs/adherents/type.php]: @hregis thank you i change it immediately

File: htdocs/adherents/subscription.php [hunk 2]
```diff
@@ -810,54 +847,103 @@
...
+$dateto = strtotime(date("Y-m-d", $dateb) . " + 1 year - 1 day");  
+}else{
+if ($conf->global->ADHERENT_SUBSCRIPTION_PRORATA == '0') {
+$next = mktime(0,00,00,$month-$conf->global->SOCIETE_SUBSCRIBE_MONTH_PRESTART,$day,$year+1);
+$date = $dateb =$today;
+}else{ 
+$next = $startcotis2; 
+if (dol_mktime(0,00,00,$conf->global->SOCIETE_SUBSCRIBE_MONTH_START,'01',$year)>$today && $datefin<$today){$date=dol_now();}else{$date = dol_mktime(0,00,00,$conf->global->SOCIETE_SUBSCRIBE_MONTH_START,'01',$year);}
+$dateb = dol_mktime(0,00,00,$conf->global->SOCIETE_SUBSCRIBE_MONTH_START,'01',$year);} 
+$dateto = strtotime(date("Y-m-d", $dateb) . " + 1 year - 1 day");
```
Comment (eldy) [inline htdocs/adherents/subscription.php]: use the method   dol_timeplusduree() to make addition and substraction on date.

File: htdocs/adherents/subscription.php [hunk 3]
```diff
@@ -810,54 +847,103 @@
...
+$cotisationfin2 = strftime("%m",$datefin);
+$cotisationfin3 = strftime("%d",$datefin);
+
+$startcotis0 = dol_mktime(0,00,00,$conf->global->SOCIETE_SUBSCRIBE_MONTH_START-$conf->global->SOCIETE_SUBSCRIBE_MONTH_PRESTART,'01',$year-1);
+$startcotis1 = dol_mktime(0,00,00,$conf->global->SOCIETE_SUBSCRIBE_MONTH_START-$conf->global->SOCIETE_SUBSCRIBE_MONTH_PRESTART,'01',$year);
+$startcotis2 = dol_mktime(0,00,00,$conf->global->SOCIETE_SUBSCRIBE_MONTH_START-$conf->global->SOCIETE_SUBSCRIBE_MONTH_PRESTART,'01',$year+1);
+
+if ($startcotis1>$today){
+if ($conf->global->ADHERENT_SUBSCRIPTION_PRORATA == '0') { 
+$next = mktime(5,00,00,$month-$conf->global->SOCIETE_SUBSCRIBE_MONTH_PRESTART,$day,$year);
```
Comment (eldy) [inline htdocs/adherents/subscription.php]: What is the 5 for ?

Comment (ptibogxiv) [reply htdocs/adherents/subscription.php]: @eldy bug with my timeserver while coding,  must be 00 of course

File: htdocs/adherents/subscription.php (hunk 2 above)
Comment (ptibogxiv) [reply htdocs/adherents/subscription.php]: @eldy fix done ;-)

**Comment by eldy** [comment]
I noticed several different features in this PR. It is difficult to me to validate. Is it possible to push one feature at time.
For example, mu understanding is that the type of member is now saved into the subscription (this allow to know what was the type if the member change its type the next year i suppose). If i am right, this can be one PR.

I am not sure to have understand goal of other new field
family, automatic, welcome

**Pull request closed without merging**